api/query/query.go

```diff
@@ -61,6 +61,30 @@ func (qh queryHandler) processInput(w http.ResponseWriter, r *http.Request) (*sh
 	return &filters, testRuns, summaries, nil
 }
 
+func (qh queryHandler) getVersionFiles(w http.ResponseWriter, r *http.Request) ([]string, error) {
+
+	filters, err := shared.ParseQueryFilterParams(r.URL.Query())
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return nil, err
+	}
+	testRuns, filters, err := qh.getRunsAndFilters(filters)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return nil, err
+	}
+	versionFiles := make([]string, 0, len(testRuns))
+	for _, testRun := range testRuns {
+		summaryURL := shared.GetResultsURL(testRun, "")
+		url := summaryURL[:len(summaryURL)-16] + "_version.txt"
+		mkey := getRedisKey(testRun)
+		var data []byte
+		qh.dataSource.Get(mkey, url, &data)
```

Comment:
should there be an error check here? if the error doesn't need to be returned up, just log a warning

Reply:
I'll add a debug log here - it's not unlikely that there should be an error fetching this file and it simply means that this is associated with an old summary file and we should disregard that file.